### PR TITLE
refactor(meta): remove single parallel unit and `ParallelUnitType`

### LINKED ISSUE #4109: refactor: remove single ParallelUnit
Now we have 2 kinds of parallel units: hash and single. Single ParallelUnit was designed to schedule single fragments (e.g. the global aggregation operator) on. However, it also increases the complexity because there are 2 kinds of parallel units now.

I think it is unnecessary to introduce such concept. Thus, all parallel units are the current hash parallel units, and single fragments can be simply scheduled to a **random** parallel unit.

### PR DESCRIPTION
**Pull request by HuaHuaY**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

Remove single parallel unit and `ParallelUnitType`.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)
close #4109

### CODE REVIEW

File: src/meta/src/cluster/mod.rs [hunk 1]
```diff
@@ -350,8 +332,7 @@ pub struct ClusterManagerCore {
     /// Record for workers in the cluster.
     workers: HashMap<WorkerKey, Worker>,
 
-    /// Record for parallel units of different types.
-    single_parallel_units: Vec<ParallelUnit>,
+    /// Record for parallel units.
     hash_parallel_units: Vec<ParallelUnit>,
```
Comment (fuyufjh) [inline src/meta/src/cluster/mod.rs:336]:
```suggestion
    parallel_units: Vec<ParallelUnit>,
```

Comment (fuyufjh) [reply src/meta/src/cluster/mod.rs:336]: Please also check other occurrences

`s/hash_parallel_units/parallel_units/g`

**Review by fuyufjh — APPROVED** [review]
LGTM for rest

File: proto/common.proto [hunk 2]
```diff
@@ -32,14 +32,8 @@ enum WorkerType {
...
 
-enum ParallelUnitType {
-  SINGLE = 0;
-  HASH = 1;
-}
-
 message ParallelUnit {
   uint32 id = 1;
-  ParallelUnitType type = 2;
   uint32 worker_node_id = 3;
```
Comment (BugenZhao) [inline proto/common.proto:37]:
```suggestion
  uint32 worker_node_id = 2;
```

File: src/frontend/src/scheduler/distributed/query.rs [hunk 3]
```diff
@@ -525,15 +523,10 @@ mod tests {
...
     fn generate_parallel_units(start_id: u32, node_id: u32) -> Vec<ParallelUnit> {
         let parallel_degree = 8;
-        let mut parallel_units = vec![ParallelUnit {
-            id: start_id,
-            r#type: ParallelUnitType::Single as i32,
-            worker_node_id: node_id,
-        }];
-        for id in start_id + 1..start_id + 1 + parallel_degree {
+        let mut parallel_units = vec![];
+        for id in start_id..start_id + parallel_degree {
```
Comment (BugenZhao) [inline src/frontend/src/scheduler/distributed/query.rs:527]: Or `Vec::with_capacity`.
```suggestion
        (start_id..start_id + parallel_degree).map(..).collect()
```

File: src/frontend/src/scheduler/plan_fragmenter.rs [hunk 4]
```diff
@@ -780,15 +778,10 @@ mod tests {
...
     fn generate_parallel_units(start_id: u32, node_id: u32) -> Vec<ParallelUnit> {
         let parallel_degree = 8;
-        let mut parallel_units = vec![ParallelUnit {
-            id: start_id,
-            r#type: ParallelUnitType::Single as i32,
-            worker_node_id: node_id,
-        }];
-        for id in start_id + 1..start_id + 1 + parallel_degree {
+        let mut parallel_units = vec![];
+        for id in start_id..start_id + parallel_degree {
```
Comment (BugenZhao) [inline src/frontend/src/scheduler/plan_fragmenter.rs:782]: Ditto.

File: src/meta/src/cluster/mod.rs (hunk 1 above)
Comment (BugenZhao) [inline src/meta/src/cluster/mod.rs:336]:
```suggestion
    parallel_units: Vec<ParallelUnit>,
```

File: src/meta/src/stream/meta.rs [hunk 5]
```diff
@@ -329,21 +329,10 @@ where
     ) -> Result<(Vec<TableFragments>, HashMap<ParallelUnitId, ParallelUnit>)> {
         let mut parallel_unit_migrate_map = HashMap::new();
         let mut pu_hash_map: HashMap<WorkerId, Vec<&ParallelUnit>> = HashMap::new();
```
Comment (BugenZhao) [inline src/meta/src/stream/meta.rs:331]:
```suggestion
        let mut pu_map: HashMap<WorkerId, Vec<&ParallelUnit>> = HashMap::new();
```

File: src/meta/src/stream/scheduler.rs [hunk 6]
```diff
@@ -173,19 +168,12 @@ where
...
-                        .await;
-                    let single_idx = self
-                        .single_rr
-                        .fetch_update(Ordering::SeqCst, Ordering::SeqCst, |idx| {
-                            Some((idx + 1) % single_parallel_units.len())
-                        })
-                        .unwrap();
-
-                    single_parallel_units[single_idx].clone()
+                    // Choose one parallel unit to schedule from hash parallel units.
```
Comment (BugenZhao) [inline src/meta/src/stream/scheduler.rs:171]:
```suggestion
                    // Randomly choose one parallel unit to schedule from all parallel units.
```

**Review by BugenZhao — COMMENTED** [review]
Rest LGTM. Please cleanup `hash` words.

**Review by yezizp2012 — APPROVED** [review]
Generally LGTM, except the naming with hash.

**Pull request merged**